engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1828,6 +1796,104 @@ public void run()
     return rsp;
   }
 
+  static class UpdateOperatorLatencyContext
+  {
+    StreamingContainerManager scm;
+
+    UpdateOperatorLatencyContext()
+    {
+    }
+
+    UpdateOperatorLatencyContext(StreamingContainerManager scm)
+    {
+      this.scm = scm;
+    }
+
+    long getRPCLatency(PTOperator oper)
+    {
+      MovingAverageLong rpcLatency = this.scm.rpcLatencies.get(oper.getContainer().getExternalId());
+      return rpcLatency == null ? 0 : rpcLatency.getAvg();
+    }
+
+    boolean endWindowStatsExists(long windowId)
+    {
+      return scm.endWindowStatsOperatorMap.containsKey(windowId);
+    }
+
+    long getEndWindowEmitTimestamp(long windowId, PTOperator oper)
+    {
+      Map<Integer, EndWindowStats> endWindowStatsMap = scm.endWindowStatsOperatorMap.get(windowId);
+      if (endWindowStatsMap == null) {
+        return -1;
+      }
+      EndWindowStats ews = endWindowStatsMap.get(oper.getId());
+      if (ews == null) {
+        return -1;
+      }
+      return ews.emitTimestamp;
+    }
+  }
+
+  public long updateOperatorLatency(PTOperator oper, UpdateOperatorLatencyContext ctx)
```

Comment:
I think we should keep it the way it is for the scope of this PR since it is consistent with the existing checkpoint-related code  in StreamingContainerManager. We can open a separate ticket to refactor the code to make this class smaller though.